### DISCUSSION TITLE
Execute the same job multiple times at the same

### DISCUSSION THREAD

**Discussion by linkdotnet** [NCronJob-Dev/NCronJob#79] [Polls]
Given a job registration as the following:
```csharp
b.WithCronExpression("0 * * * *").WithParameter(100)
 .And
 .WithCronExpression("0 * * * *").WithParameter(200)
```

This seems like a valid case - I want to have a job that gets triggered every hour two times but with a different set of parameters.

If we generalize from here, should this be valid?
```csharp
b.WithCronExpression("0 * * * *")
 .And
 .WithCronExpression("0 * * * *")
```

That said: The sample given above: Should this be one or two executions at the same time?

**Comment by falvarez1** [comment]
To me, it makes more sense to treat identical schedules and jobs as a single registry entry. The system should log a visible warning indicating that only the first schedule entry will be used, along with a detailed reason why this decision was made. Only if the name, parameters, etc., are different should each have its own entry. My reasoning is that if concurrency is not set and you have a job type with multiple identical schedules, you end up with multiple objects that are essentially the same and can only run one at a time. In this instance, the system will oscillate between each registry entry even though the job is exactly the same. This can lead to inefficient resource utilization and potential confusion.

That said, this behavior could be controlled by a configuration setting with a sensible default. Once we finalize the configuration, this could be categorized as an `advanced` setting.

**Reply by linkdotnet** [reply]
I do like the point you are making.
My initial reasoning was:
 1. It keeps our code easier
 2. It is something the user explicitly configured
 
For (1) we could easily just use `HashSet` or to "counter" the behavior without onboarding complexity.
Logging might be something, paired with good documentation that highlights that specific edge case.

If a user really wants two executions, this is still possible. Just give two different job names.

**Reply by falvarez1** [reply]
> If a user really wants two executions, this is still possible. Just give two different job names.

My thoughts exactly
